### PR TITLE
Fixes #37437 - Prevent XSS issue for pages using react and angular

### PR DESCRIPTION
**Pull request by sbernhard**
Additionally, it prevents to provide all user data to react.

### CODE REVIEW

**Comment by sbernhard** [comment]
I dont think the test failures are releated.

**Comment by jeremylenz** [comment]
Checking to make sure our JS code does not rely on any of the data being removed here:

This data is exposed in React via the `useForemanUser` hook. I don't see any actual uses of that in the Foreman or Katello orgs (I also checked for `useForemanContext` and `useForemanMetadata`), so I think we're okay. LGTM :+1:

File: app/helpers/layout_helper.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def fetch_locations
   end
 
   def fetch_user
-    { current_user: User.current, user_dropdown: Menu::Manager.to_hash(:side_menu), impersonated_by: User.unscoped.find_by_id(session[:impersonated_by]) }
+    { current_user: User.current.to_json(only: [:id, :firstname, :lastname, :mail, :admin, :last_login_on, :name]), user_dropdown: Menu::Manager.to_hash(:side_menu), impersonated_by: User.unscoped.find_by_id(session[:impersonated_by]) }
```
Comment (MariaAga) [inline app/helpers/layout_helper.rb]: should be as_json to not break Layout.js

Comment (sbernhard) [reply app/helpers/layout_helper.rb]: Thx!

**Review by MariaAga — APPROVED** [review]

**Comment by MariaAga** [comment]
Merged, Thanks!

**Pull request merged**